source/mir/random/generic/types.d

```diff
@@ -0,0 +1,232 @@
+module mir.random.generic.types;
+
+import std.traits: ReturnType;
+
+protected:
+
+/**
+Major data unit of the Tinflex algorithm.
+It is used to store
+- (cached) values of the transformation (and its derivatives)
+- area below the hat and squeeze function
+- linked-list like reference to the right part of the interval (there will always
+be exactly one interval with right = 0)
+*/
+struct IntervalPoint(S)
```

Comment:
A user may do not know f1 and f2, but it know their values in required points. An API should support this